sdk/node/Libplanet.Node/Options/NodeOptionsBase.cs

```diff
@@ -0,0 +1,16 @@
+using System.ComponentModel;
+using Libplanet.Node.DataAnnotations;
+
+namespace Libplanet.Node.Options;
+
+public abstract class NodeOptionsBase<T> : OptionsBase<T>
+    where T : NodeOptionsBase<T>
+{
+    [PrivateKey]
+    [Description("The private key.")]
+    public string PrivateKey { get; set; } = string.Empty;
+
+    [AppProtocolVersion]
+    [Description("The version of the application protocol.")]
+    public string AppProtocolVersion { get; set; } = string.Empty;
```

Comment:
Private key and APV are not options that depend on the node itself, but on Swarm. This node will not need Private key or APV if it runs solo, but the name NodeBase feels too verbose.

Reply:
> Private key and APV are not options that depend on the node itself, but on Swarm. This node will not need Private key or APV if it runs solo, but the name NodeBase feels too verbose.

How about `SwarmOptionsBase` or below are some names suggested by github copilot.

* NetworkOptionsBase
* ProtocolOptionsBase
* CommonNodeOptions
* SharedNodeOptions
* NodeProtocolOptions

Reply:
IMO `NetworkOptionsBase` or `ProtocolOptions` is much better in this case.

Reply:
> IMO `NetworkOptionsBase` or `ProtocolOptions` is much better in this case.

Github copilot recommended `ProtocolOptionsBase` for the examples.

Reply:
Finally, `AppProtocolOptionsBase` was selected. https://github.com/planetarium/libplanet/pull/3932/commits/369215639017c881755d19193b60f9c7b62fddb3